File: storage/db-core/src/main/java/com/coffee/gu/menu/MenuOptionGroupEntity.java
package com.coffee.gu.menu;

import com.coffee.gu.BaseEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;

@Table(name = "menu_option_group")
@Entity
public class MenuOptionGroupEntity extends BaseEntity {

    private Long menuId;
    private Long optionGroupId;

    public MenuOptionGroupEntity() {}

    public MenuOptionGroupEntity(Long menuId, Long optionGroupId) {
        this.menuId = menuId;
        this.optionGroupId = optionGroupId;
    }

    public  MenuOptionGroup toModel() {
        return new MenuOptionGroup(
                menuId,
                optionGroupId
        );
    }

    public Long getMenuId() {
        return menuId;
    }

    public Long getOptionGroupId() {
        return optionGroupId;
    }
}

File: storage/db-core/src/main/kotlin/com/coffee/gu/menu/MenuOptionGroupEntity.kt
package com.coffee.gu.menu

import com.coffee.gu.BaseEntity
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table

@Table(name = "menu_option_group")
@Entity
class MenuOptionGroupEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0,
    val menuId: Long,
    val optionGroupId: Long,
) : BaseEntity() {

    fun toModel(): MenuOptionGroup {
        return MenuOptionGroup(
            menuId = menuId,
            optionGroupId = optionGroupId,
        )
    }
}
